snapcraft/tests/repo/test_snaps.py

```diff
@@ -0,0 +1,377 @@
+              expected=True)),
+        ('installed stable with channel',
+         dict(snap='fake-snap-stable/latest/stable',
+              expected=True)),
+        ('not installed',
+         dict(snap='missing-snap',
+              expected=False)),
+        ('not installed with channel',
+         dict(snap='missing-snap/latest/stable',
+              expected=False)),
+    ]
+
+    def test_is_installed(self):
+        snap_pkg = snaps.SnapPackage(self.snap)
+        self.assertThat(snap_pkg.installed, Is(self.expected))
+
+    def test_is_installed_classmethod(self):
+        self.assertThat(snaps.SnapPackage.is_snap_installed(self.snap),
+                        Is(self.expected))
+
+
+class SnapPackageIsInStoreTest(SnapPackageBaseTestCase):
+
+    scenarios = [
+        ('in store',
+         dict(snap='fake-snap',
+              expected=True)),
+        ('in store with channel',
+         dict(snap='fake-snap/latest/stable',
+              expected=True)),
+        ('not in store',
+         dict(snap='missing-snap',
+              expected=False)),
+        ('not in store with channel',
+         dict(snap='missing-snap/latest/stable',
+              expected=False)),
+    ]
+
+    def test_is_in_store(self):
+        snap_pkg = snaps.SnapPackage(self.snap)
+        self.assertThat(snap_pkg.in_store, Is(self.expected))
+
+
+class SnapPackageIsClasicTest(SnapPackageBaseTestCase):
+
+    scenarios = [
+        ('classic', dict(snap='fake-snap/classic/stable', expected=True)),
+        ('strict', dict(snap='fake-snap/strict/stable', expected=False)),
+        ('devmode', dict(snap='fake-snap/devmode/stable', expected=False)),
+    ]
+
+    def test_is_classic(self):
+        snap_pkg = snaps.SnapPackage(self.snap)
+        self.assertThat(snap_pkg.is_classic(), Is(self.expected))
+
+
+class SnapPackageIsValidTest(SnapPackageBaseTestCase):
+
+    scenarios = [
+        ('valid',
+         dict(snap='fake-snap', expected=True)),
+        ('valid with channel',
+         dict(snap='fake-snap/strict/stable', expected=True)),
+        ('invalid',
+         dict(snap='missing-snap', expected=False)),
+        ('invalid with channel',
+         dict(snap='missing-snap/strict/stable', expected=False)),
+
+    ]
+
+    def test_is_valid(self):
+        snap_pkg = snaps.SnapPackage(self.snap)
+        self.assertThat(snap_pkg.is_valid(), Is(self.expected))
+
+    def test_is_valid_classmethod(self):
+        self.assertThat(snaps.SnapPackage.is_valid_snap(self.snap),
+                        Is(self.expected))
+
+
+class SnapPackageLifecycleTest(SnapPackageBaseTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.fake_snap_command = FakeSnapCommand()
+        self.useFixture(self.fake_snap_command)
+
+    def test_install_classic(self):
+        snap_pkg = snaps.SnapPackage('fake-snap/classic/stable')
+        snap_pkg.install()
+        self.assertThat(self.fake_snap_command.calls, Equals([
+            ['snap', 'whoami'],
```

Comment:
There's a test missing for the case of a logged in user where `sudo` won't be used.